server/src/logger.rs

```diff
@@ -59,41 +60,18 @@ impl std::fmt::Display for MethodKind {
 	}
 }
 
-/// Defines a logger specifically for HTTP requests with callbacks during the RPC request life-cycle.
-/// The primary use case for this is to collect timings for a larger metrics collection solution.
-///
-/// See [`HttpServerBuilder::set_logger`](../../jsonrpsee_http_server/struct.HttpServerBuilder.html#method.set_logger) method
-/// for examples.
-pub trait HttpLogger: Send + Sync + Clone + 'static {
-	/// Intended to carry timestamp of a request, for example `std::time::Instant`. How the trait
-	/// measures time, if at all, is entirely up to the implementation.
-	type Instant: std::fmt::Debug + Send + Sync + Copy;
-
-	/// Called when a new JSON-RPC request comes to the server.
-	fn on_request(&self, remote_addr: SocketAddr, request: &Request<Body>) -> Self::Instant;
-
-	/// Called on each JSON-RPC method call, batch requests will trigger `on_call` multiple times.
-	fn on_call(&self, method_name: &str, params: Params, kind: MethodKind);
-
-	/// Called on each JSON-RPC method completion, batch requests will trigger `on_result` multiple times.
-	fn on_result(&self, method_name: &str, success: bool, started_at: Self::Instant);
-
-	/// Called once the JSON-RPC request is finished and response is sent to the output buffer.
-	fn on_response(&self, result: &str, _started_at: Self::Instant);
-}
-
 /// Defines a logger specifically for WebSocket connections with callbacks during the RPC request life-cycle.
 /// The primary use case for this is to collect timings for a larger metrics collection solution.
 ///
-/// See the [`WsServerBuilder::set_logger`](../../jsonrpsee_ws_server/struct.WsServerBuilder.html#method.set_logger)
+/// See the [`ServerBuilder::set_logger`](../../jsonrpsee_server/struct.ServerBuilder.html#method.set_logger)
 /// for examples.
-pub trait WsLogger: Send + Sync + Clone + 'static {
+pub trait Logger: Send + Sync + Clone + 'static {
 	/// Intended to carry timestamp of a request, for example `std::time::Instant`. How the trait
 	/// measures time, if at all, is entirely up to the implementation.
 	type Instant: std::fmt::Debug + Send + Sync + Copy;
 
 	/// Called when a new client connects
-	fn on_connect(&self, remote_addr: SocketAddr, headers: &Headers);
+	fn on_connect(&self, _remote_addr: Option<SocketAddr>, _request: &HttpRequest) {}
 
 	/// Called when a new JSON-RPC request comes to the server.
 	fn on_request(&self) -> Self::Instant;
```

Comment:
Is there any reason we wouldn't want to give everything here except `on_request` a default impl?

(reasoning being; people can choose to log whatever they are interested in and ignore the rest that was)

Reply:
I think I had `HTTP` vs `WS` in mind but that doesn't make sense anymore